lib/va_profile/profile_information/service.rb

```diff
@@ -0,0 +1,201 @@
+        address: 'Address',
+        residence_address: 'Home address',
+        correspondence_address: 'Mailing address',
+        email: 'Email address',
+        phone: 'Phone number',
+        home_phone: 'Home phone number',
+        mobile_phone: 'Mobile phone number',
+        work_phone: 'Work phone number'
+      }.freeze
+
+      # def get_response(type)
+      #   with_monitoring do
+      #     icn_with_aaid_present!
+      #     model = "VAProfile::Models::#{type.capitalize}".constantize
+      #     raw_response = perform(:post, path, { bios: [{ bioPath: 'bio' }] })
+      #     response = model.response_class(raw_response)
+      #     Sentry.set_extras(response.debug_data) unless response.ok?
+      #     response
+      #   end
+      # rescue Common::Client::Errors::ClientError => e
+      #   if e.status == 404
+      #     log_exception_to_sentry(
+      #       e,
+      #       { vet360_id: @user.vet360_id },
+      #       { va_profile: :person_not_found },
+      #       :warning
+      #     )
+      #     return PersonResponse.new(404, person: nil)
+
+      #   elsif e.status >= 400 && e.status < 500
+      #     return PersonResponse.new(e.status, person: nil)
+      #   end
+      #   handle_error(e)
+      # rescue => e
+      #   handle_error(e)
+      # end
+
+      def get_person
+        with_monitoring do
+          vet360_id_present!
+          raw_response = perform(:get, @user.vet360_id)
+
+          PersonResponse.from(raw_response)
+        end
+      rescue Common::Client::Errors::ClientError => e
+        if e.status == 404
+          log_exception_to_sentry(
+            e,
+            { vet360_id: @user.vet360_id },
+            { va_profile: :person_not_found },
+            :warning
+          )
+
+          return PersonResponse.new(404, person: nil)
+        elsif e.status >= 400 && e.status < 500
+          return PersonResponse.new(e.status, person: nil)
+        end
+
+        handle_error(e)
+      rescue => e
+        handle_error(e)
+      end
+
+      def self.get_person(vet360_id)
+        stub_user = OpenStruct.new(vet360_id:)
+        new(stub_user).get_response('person')
+      end
+
+      def submit(params)
+        config.submit(path(@user.edipi), params)
+      end
+
+      # Record is not defined when requesting an #update
+      # Determine if the record needs to be created or updated with reassign_http_verb
+      # Ensure http_verb is a symbol for the response request
+      def create_or_update_info(http_verb, record)
+        with_monitoring do
+          icn_with_aaid_present!
+          model = record.class
+          http_verb = http_verb.to_sym == :update ? reassign_http_verb(record) : http_verb.to_sym
+          update_path = record.try(:permission_type).present? ? 'permissions' : record.contact_info_attr.pluralize
+          raw_response = perform(http_verb, update_path, record.in_json)
+          response = model.transaction_response_class.from(raw_response)
```

Comment:
Given that this implementation relies on duck typing and the presence of specific methods on the models, etc, Could you explicitly call out these methods as required in the base class (via comments or something). This ensures that all subclasses implement these necessary methods for the functionality to work correctly. The base class itself doesn't enforce how these methods are implemented, only that they must exist.

Reply:
This is due to the create_or_update action being called to vets-api. In this method the create_or_update is reassigned in the reassign_http_verb method. the reassign_http_verb looks to see if the record exists or does not exists.